### PR TITLE
Map Structure PalettedBlockInfoList

### PR DESCRIPTION
**Pull request by liach**
Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Review by Runemoro — APPROVED** [review]

**Comment by Runemoro** [comment]
PR looks good, but maybe we should find a better name than "block info".

**Comment by liach** [comment]
It holds a list of structure block infos (each is like an info about a block in a structure). The Structure class holds a list of these lists. These lists have the same block arrangements, but they have different palettes for each arrangement (i.e. in one palette the block may be stone, the other blackstone)

File: mappings/net/minecraft/structure/Structure.mapping [hunk 1]
```diff
@@ -1,5 +1,5 @@
 CLASS net/minecraft/class_3499 net/minecraft/structure/Structure
-	FIELD field_15586 blocks Ljava/util/List;
+	FIELD field_15586 blockInfoLists Ljava/util/List;
```
Comment (Juuxel) [inline mappings/net/minecraft/structure/Structure.mapping:2]: Is this like `List<List<...>>`?

Comment (liach) [reply mappings/net/minecraft/structure/Structure.mapping:2]: Effectively yes, but it is `List<PalettedBlockInfoList>`, where `PalettedBlockInfoList` contains a `List<StructureBlockInfo>`, so it's like `List<List<StructureBlockInfo>>`

**Review by Juuxel — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Pull request merged**